content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -1,8 +1,136 @@
+operation.
+
+Additionally, there are two special files created by certain
+[DVC commands](/doc/command-reference):
+
+- Files ending with the `.dvc` extension are basic placeholders to version data
+  files and directories. A <abbr>DVC project</abbr> can have multiple
+  [`.dvc` files](#dvc-files).
+- The [`dvc.yaml` file](#dvcyaml-file) or _pipeline(s) file_ specifies stages
+  that form the pipeline(s) of a project, and their connections (_dependency
+  graph_ or DAG).
+
+Both use human-friendly YAML schemas, described below. We encourage you to get
+familiar with them so you may edit them freely, as needed. Both type of files
+should be versioned with Git (for Git-enabled <abbr>repositories</abbr>).
+
+## .dvc files
+
+When you add a file or directory to a <abbr>DVC project</abbr> with `dvc add` or
+`dvc import`, a `.dvc` file is created based on the data file name (e.g.
+`data.xml.dvc`). These files contain the basic information needed to track the
+data with DVC.
+
+They use a simple YAML format that can be easily written or altered manually.
+Here is a sample:
+
+```yaml
+outs:
+  - md5: a304afb96060aad90176268345e10355
+    path: data.xml
+# Manual comments can be added in.
+```
+
+`.dvc` files contain a single top field:
+
+- `outs` - list of <abbr>outputs</abbr> for this `.dvc` file
+
+An output entry can consist of these fields:
+
+- `md5` - hash value for the output file
+- `path` - path to the output in the <abbr>workspace</abbr>, relative to the
+  location of the `.dvc` file
+- `cache` - (optional) whether or not DVC should cache the output. `true` by
+  default
+
+Note that comments can be added to `.dvc` files and `dvc.yaml` using the
+`# comment` syntax.
+
+> `.dvc` file comments are preserved among executions of the `dvc repro` and
+> `dvc commit` commands, but not when a `.dvc` file is overwritten by
+> `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml file
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  stageone:
+    cmd: python cmd.py input.data output.data metrics.json
+    deps:
+    - cmd.py
+    - input.data
+    outs:
+    - output.data
+    metrics:
+    - metrics.json
+  stagetwo:
+    cmd: python ...
+    ...
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`. Each stage can contain
+the following fields:
+
+- `cmd` - executable command defined in this stage
+- `deps` - list of <abbr>dependencies</abbr> for this stage
+- `params` - (optional) list of the [parameter](/doc/command-reference/params)
+  names and their current values
+- `outs` - list of <abbr>outputs</abbr> for this stage
+- `metric` - (optional) list of [metric](/doc/command-reference/metrics) files
+- `locked` - (optional) whether or not this stage is locked from reproduction
```

Comment:
```suggestion
- `frozen` - (optional) whether or not this stage is locked from reproduction
```

Reply:
OK updated here and in the import cmd ref in 8fbf1294. Thanks